.github/workflows/cd.yaml

```diff
@@ -0,0 +1,69 @@
+name: Deploy to ECS
+
+on:
+  push:
+    branches:
+      - main
+
+env:
+  AWS_REGION: eu-central-1
+  ECR_REPOSITORY_NAME: cloud-auth
+  ECS_CLUSTER_NAME: cloud-auth
+  ECS_SERVICE_NAME: cloud-auth-service
+  TASK_FAMILY_NAME: cloud-auth
+  PORT: 3333
+  CLOUD_APP_ORIGIN: ${{ secrets.CLOUD_APP_ORIGIN }}
+  SUPABASE_JWT_SECRET: ${{ secrets.SUPABASE_JWT_SECRET }}
+  COOKIE_NAME: ${{ secrets.COOKIE_NAME }}
+  COOKIE_SECRET: ${{ secrets.COOKIE_SECRET }}
+  DIRECT_URL: ${{ secrets.DIRECT_URL }}
+  DATABASE_URL: ${{ secrets.DATABASE_URL }}
+
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Set up Terraform
+        uses: hashicorp/setup-terraform@v1
+        with:
+          terraform_version: 1.1.0
+
+      - name: Initialize Terraform
+        run: terraform init
+
+      - name: Log in to Amazon ECR
+        id: login-ecr
+        uses: aws-actions/amazon-ecr-login@v1
+        with:
+          region: ${{ env.AWS_REGION }}
+
+      - name: Build and push Docker image to Amazon ECR
+        run: |
+          docker build -t ${env.ECR_REPOSITORY_NAME}:${{ github.sha }} .
+          docker tag ${env.ECR_REPOSITORY_NAME}:${{ github.sha }} ${env.ACCOUNT_ID}.dkr.ecr.${env.AWS_REGION}.amazonaws.com/${env.ECR_REPOSITORY_NAME}:${{ github.sha }}
+          docker push ${env.ACCOUNT_ID}.dkr.ecr.${env.AWS_REGION}.amazonaws.com/${env.ECR_REPOSITORY_NAME}:${{ github.sha }}
+
+      - name: Deploy to Amazon ECS
+        uses: hashicorp/terraform-github-actions@master
+        with:
+          tf_actions_version: "0.2.0"
+          working-directory: ./terraform
+          terraform-actions-command: "apply"
+          args: |
```

Comment:
You can use env vars for the Terraform variables, https://developer.hashicorp.com/terraform/cli/config/environment-variables#tf_var_name